src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1171,6 +1171,16 @@ public function getNormalizedIds()
         return $normalizedIds;
     }
 
+    public function logRemoveService(CompilerPassInterface $pass, $id, $reason)
+    {
+        $this->getCompiler()->log($pass, sprintf('Removed service "%s"; reason: %s.', $id, $reason));
+    }
+
+    public function log(CompilerPassInterface $pass, $message, $level = E_USER_NOTICE)
```

Comment:
I don't think passing a level here makes much sense, especially not as a PHP error level. It would couple it too much to the error triggering layer, while we talk about logging here.
If you want to provide a logging level, use PSR-3 levels instead.

Btw, a way to allow flexibility about adding a level only the day we need it is to make the method final: final method can add optional args as they want (adding an optional arg only breaks BC for inheritance)